cli/src/cli.rs

```diff
@@ -471,11 +471,15 @@ impl CliConfig<'_> {
             (SettingType::Explicit, websocket_cfg_url.to_string()),
             (
                 SettingType::Computed,
-                solana_cli_config::Config::compute_websocket_url(json_rpc_cmd_url),
+                solana_cli_config::Config::compute_websocket_url(&normalize_to_url_if_moniker(
+                    json_rpc_cmd_url,
+                )),
             ),
             (
                 SettingType::Computed,
-                solana_cli_config::Config::compute_websocket_url(json_rpc_cfg_url),
+                solana_cli_config::Config::compute_websocket_url(&normalize_to_url_if_moniker(
+                    json_rpc_cfg_url,
+                )),
```

Comment:
This change enables users to store `json_rpc_url: d` or similar in the config file and parse properly (without those scary panics), so technically https://github.com/solana-labs/solana/pull/14981/commits/c3ab909d882b1c0a75cc9315eb80a799188713e0 isn't required. However, I prefer storing the verbose url in the config file instead of the moniker.